### PR TITLE
[Hexagon] Add test for registered schedules

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR adds test for registered schedules on hexagon target. Follow up to this PR: https://github.com/apache/tvm/pull/10919

cc @kparzysz-quic @csullivan @Lunderberg

### CODE REVIEW

**Comment by mehrdadh** [comment]
@farshidsp

File: tests/scripts/task_build_hexagon_api.sh [hunk 1]
```diff
@@ -19,8 +19,18 @@
 set -e
 set -u
 
+use_cache=false
+if [ $# -ge 1 ] && [[ "$1" == "--use-cache" ]]; then
+    use_cache=true
+    shift 1
+fi
+
```
Comment (csullivan) [inline tests/scripts/task_build_hexagon_api.sh:27]: Unrelated?

File: tests/python/contrib/test_hexagon/test_batch_matmul.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (csullivan) [inline tests/python/contrib/test_hexagon/test_batch_matmul.py:1]: Should we move these tests to an `topi/` subdirectory? Thinking about how we might want the directory structure. These are generic functionality tests, but soon we'll have tests for some of the same operators which utilize more complex scheduling. It would be nice to have these located in separate directories.

**Review by csullivan — COMMENTED** [review]
This all looks great to me, just a couple comments below for discussion

File: python/tvm/script/tir/__init__.pyi [hunk 3]
```diff
@@ -226,6 +226,7 @@ def alloc_buffer(
 """
 special_stmt - Reads/Writes
 """
+
```
[the remark this answers is not in the document]
Comment (mehrdadh) [reply python/tvm/script/tir/__init__.pyi:229]: This is done by the clang format tool.

File: python/tvm/script/tir/__init__.pyi [hunk 4]
```diff
@@ -337,6 +338,7 @@ def Assert(condition: Union[PrimExpr, builtins.bool], message: str) -> PrimExpr:
 """
 Scope handler - Loops
 """
+
```
[the remark this answers is not in the document]
Comment (mehrdadh) [reply python/tvm/script/tir/__init__.pyi:341]: same as above.

File: src/runtime/hexagon/rpc/hexagon/rpc_server.cc [hunk 5]
```diff
@@ -41,7 +41,7 @@ extern "C" {
 #include "hexagon_rpc.h"
 
 // TODO(mehrdadh): make this configurable.
-#define TVM_HEXAGON_RPC_BUFF_SIZE_BYTES 2 * 1024 * 1024
+#define TVM_HEXAGON_RPC_BUFF_SIZE_BYTES 5 * 1024 * 1024
```
[the remark this answers is not in the document]
Comment (mehrdadh) [reply src/runtime/hexagon/rpc/hexagon/rpc_server.cc:44]: yeah makes sense. Waiting for https://github.com/apache/tvm/pull/11022 to merge and then will rebase and fix this.

File: tests/scripts/task_build_hexagon_api.sh (hunk 1 above)
Comment (mehrdadh) [reply tests/scripts/task_build_hexagon_api.sh:27]: Unrelated to this PR but useful change for development. I could send a separate PR if you suggest.

File: tests/python/contrib/test_hexagon/test_batch_matmul.py (hunk 2 above)
Comment (mehrdadh) [reply tests/python/contrib/test_hexagon/test_batch_matmul.py]: makes sense. Moved them to sub-directories.

**Review by csullivan — APPROVED** [review]
:shipit:
